File: src/main/java/uk/yermak/audiobookconverter/ConverterOutput.java
package uk.yermak.audiobookconverter;

public class ConverterOutput {
    private MediaInfo mediaInfo;
    private final String outputFileName;

    public MediaInfo mediaInfo() {
        return this.mediaInfo;
    }

    public void mediaInfo_$eq(final MediaInfo x$1) {
        this.mediaInfo = x$1;
    }

    public String outputFileName() {
        return this.outputFileName;
    }

    public String getOutputFileName() {
        return this.outputFileName();
    }

    public long getDuration() {
        return this.mediaInfo().getDuration();
    }

    public MediaInfo getMediaInfo() {
        return this.mediaInfo();
    }

    public ConverterOutput(final MediaInfo mediaInfo, final String outputFileName) {
        this.mediaInfo = mediaInfo;
        this.outputFileName = outputFileName;
    }
}

        

File: src/main/scala/uk/yermak/audiobookconverter/ConverterOutput.scala
package uk.yermak.audiobookconverter

/**
  * Created by Yermak on 29-Dec-17.
  */
class ConverterOutput(var mediaInfo: MediaInfo, val outputFileName: String) {
  def getOutputFileName: String = outputFileName

  def getDuration: Long = mediaInfo.getDuration

  def getMediaInfo: MediaInfo = mediaInfo
}
